### ISSUE TITLE
PeriodicCallback similar to Tornado's

### ISSUE THREAD

**Issue by postelrich** [tiangolo/fastapi#2822]
### Description
Add a function similar to tornado's [periodiccallback](https://www.tornadoweb.org/en/stable/ioloop.html#tornado.ioloop.PeriodicCallback) except with fastapi-style syntax.

### The solution you would like

I would like it to have a `teleport_to_moon` parameter that defaults to `False`, and can be set to `True` to teleport me:

```Python
from fastapi import periodic_callback

app = FastAPI()

@periodic_callback(interval=5)  # seconds
def status():
    print("Status: OK")

status.start()

@app.on_event('shutdown')
async def shutdown():
    status.stop()
```

**Comment by falkben** [comment]
Interesting feature request. What are you using `PeriodicCallback` for now? Is it just for logging a health/status check?

**Comment by falkben** [comment]
Oh, have you tried "repeated tasks" in the fastapi-utils package?
https://fastapi-utils.davidmontague.xyz/user-guide/repeated-tasks/

**Comment by postelrich** [comment]
@falkben yes thats exactly what im looking for though it would be nice not to add sqlalchemy as a dep. Thanks.

I'm using it to poll for maintaining a cache.

**Comment by falkben** [comment]
The code needed for this feature is pretty minimal and doesn't require any additional dependencies. You could vendor it if you are opposed to adding sqlalchemy as a dependency. It's actually kind of amazing how little code it takes to do this "trick".

https://github.com/dmontagu/fastapi-utils/blob/master/fastapi_utils/tasks.py

**Comment by ghandic** [comment]
Given this adds no additional dependencies what would be the reason to not include some of these utils into the main fastapi repo? @dmontagu

**Comment by skrech** [comment]
I also think that this small utility function will be a good addition to core fastapi.
